python/spark_sklearn/grid_search.py

```diff
@@ -2,15 +2,26 @@
 Class for parallelizing GridSearchCV jobs in scikit-learn
 """
 
-from collections import Sized
+import sys
+
+from itertools import product
+from collections import Sized, Mapping, namedtuple, defaultdict, Sequence
+from functools import partial
+import warnings
+
 import numpy as np
+from scipy.stats import rankdata
 
 from sklearn.base import BaseEstimator, is_classifier, clone
-from sklearn.cross_validation import KFold, check_cv, _fit_and_score, _safe_split
-from sklearn.grid_search import BaseSearchCV, _check_param_grid, ParameterGrid, _CVScoreTuple
 from sklearn.metrics.scorer import check_scoring
+from sklearn.model_selection import KFold, check_cv, ParameterGrid
+from sklearn.model_selection._validation import _fit_and_score
+from sklearn.model_selection._search import BaseSearchCV, _check_param_grid, _CVScoreTuple
+from sklearn.utils.fixes import MaskedArray
+from sklearn.utils.metaestimators import _safe_split
 from sklearn.utils.validation import _num_samples, indexable
 
+
 class GridSearchCV(BaseSearchCV):
     """Exhaustive search over specified parameter values for an estimator, using Spark to
```

Comment:
There are still some differences in this doc when I do a diff between our GridSearchCV and theirs.  This is true for their branches and tags for 0.18, 0.18.X and 0.18.1.  Could you please do a diff and correct the differences wherever it makes sense?